activesupport/lib/active_support/core_ext/module/attribute_accessors_per_thread.rb

```diff
@@ -41,14 +41,14 @@ def thread_mattr_reader(*syms)
       raise NameError.new("invalid attribute name: #{sym}") unless /^[_A-Za-z]\w*$/.match?(sym)
       class_eval(<<-EOS, __FILE__, __LINE__ + 1)
         def self.#{sym}
-          Thread.current[:"attr_#{name}_#{sym}"]
+          Thread.current["attr_"+ name + "_#{sym}"]
```

Comment:
`#{name}` is different from `+ name +` in `class_eval`.

`#{name}` is evaluated immediately on `class_eval`, but `+ name +` is evaluated on executing method defined in `class_eval`